packages/opentelemetry-sdk-node/test/sdk.test.ts

```diff
@@ -36,6 +38,45 @@ import * as assert from 'assert';
 import { NodeSDK } from '../src';
 import * as NodeConfig from '@opentelemetry/node/build/src/config';
 import * as Sinon from 'sinon';
+import sinon = require('sinon');
+import { Resource } from '@opentelemetry/resources';
+import { awsEc2Detector } from '@opentelemetry/resource-detector-aws';
+
+import {
+  assertServiceResource,
+  assertCloudResource,
+  assertHostResource,
+} from '@opentelemetry/resources/test/util/resource-assertions';
+import {
+  BASE_PATH,
+  HEADER_NAME,
+  HEADER_VALUE,
+  HOST_ADDRESS,
+  SECONDARY_HOST_ADDRESS,
+  resetIsAvailableCache,
+} from 'gcp-metadata';
+import { resetIsAvailableCache as otherResetIsAvailableCache } from '@opentelemetry/resource-detector-gcp/node_modules/gcp-metadata';
```

Comment:
I think this is because we are using npm over something with package caching i.e yarn or pnpm? The matching dependencies were not being shared. Semvers were the same

Reply:
Not sure I understand what you mean here

Reply:
There are 2 copies of `gcp-metadata` when running these tests (devDep of this package, dep of gcp detector). Each has their own `isAvailable` cache which needs to be individually split. I do not think this will be an issue in an actual production environment because the packages would resolve to the same fileloc. Else I can resolve by removing the devDep and importing everything from `@opentelemetry/resource-detector-gcp/node_modules/gcp-metadata`

Reply:
Oh I see. I think importing everything from `@opentelemetry/resource-detector-gcp/node_modules/gcp-metadata` is a confusing dependency bug waiting to happen. What if we just add a public `resetIsAvailableCache` static method to the gcp detector? It only needs to be called in tests right?

Reply:
Moved in b295d8d